src/core/compiler.jl

```diff
@@ -83,6 +103,14 @@ function generate_assume(variable::Expr, distribution)
             indexing = mapfoldl(string, *, idcs, init = "")
             varname = Turing.VarName(vi, Symbol(csym_str), sym, indexing)
 
+            # Sanity check.
+            isdist = if isa($(distribution), Vector)
+                all(d -> isa(d, Distribution), $(distribution))
+            else
+                isa($(distribution), Distribution)
+            end
+            @assert isdist @error($(wrong_dist_errormsg))
```

Comment:
This same block of code is repeated several times. Could we please put it in a function?